docs/metadata.md

```diff
@@ -44,16 +50,13 @@ will generate the following files:
 /var/config/foo/baz
 ```
 
-Each file can either be:
-
-- a simple string (as for `foo/bar` above) in which case the file will
-  be created with the given contents and read/write (but not execute)
-  permissions for user and read permissions for group and everyone else (in octal format `0644`).
-- a map (as for `ssh/sshd_config` and `foo/baz` above) with the
-  following mandatory keys:
-  - `content`: the contents of the file.
-  - `perm`: the permissions to create the file with.
-
+The JSON file contains two types of entry :
+- directory, described with a name, permissions and entries that will be included in this directory,
+- file, with name, permissions and content.
```

Comment:
Can we wrap the field names in \` backticks please so they are formatted.

Also the field is called `perm` not `permissions`, it would be better to document the real name. It might be better to keep the previous structure, i.e. a list of key names like.

Lastly, the way this is written it reads like `name` is a field when actually it is (now) the key in the map, can that be clarified please.

How about:
```
The JSON file consists of a map from `name` to an entry object. Each entry object has the following fields:
- `content`: if present then the entry is a file. The value is a string containing the desired contents of the file.
- `entries`: if present then the entry is a directory. The value is a map from `name` to entry objects.
- `perm`: the permissions to create the file with.

The `content` and `entries` fields are mutually exclusive, it is an error to include both, one or the other _must_ be present. The file or directory's name in each case is the same as the key which referred to that entry.
```